rpc/block_svc_routines.c

```diff
@@ -4375,6 +4376,26 @@ block_create_common(blockCreate *blk, char *control, char *volServer, char *prio
   }
   reply->exit = -1;
 
+  if (GB_ALLOC_N(reply->out, 8192) < 0) {
+    goto out;
+  }
+
+  if (GB_ASPRINTF(&exec, GB_SAVECFG_NAME_CHECK, blk->block_name) == -1) {
+    goto out;
+  }
+
+  save = gbRunnerGetOutput(exec);
+  if (save && (tmp = strchr(save, ':'))) {
```

Comment:
since we are doing -cs the save variable should be having a digit in the form of string ?
Just wondering what are we doing with strchr here ? should we just scan it into a int and check if it is >0 ?

Thanks!

Reply:
```
# grep -cs '"name": "block12",' /etc/target/saveconfig.json /etc/target/saveconfig.json.temp
/etc/target/saveconfig.json:1
# grep -cs '"name": "block123",' /etc/target/saveconfig.json /etc/target/saveconfig.json.temp
/etc/target/saveconfig.json:0
```
It will be like this.

And the code below will make sure that the output is not 0:

`if (atol(tmp + 1)) {`

Is this what you wanted ?

Or something others.
Thanks.

Reply:
gotcha!!
I thought we don't get a file name, and we just get a number.

@lxbsz please use '-h, --no-filename' option with grep to void filename. Thanks!

Reply:
Done @pkalever 
Thanks.